src/misc.jl

```diff
@@ -86,6 +105,11 @@ function levelsmap{T}(a::AbstractArray{T})
     return d
 end
 
+"""
+    findat!(r::IntegerArray, a::AbstractArray, b::AbstractArray)
+
+For each element in `b`, find its first index in `a`. If the value does not appear in `a`, the corresponding index is `0`. Write the results of findat(a, x) to a pre-allocated array `r`.
```

Comment:
I would add "and store the result in `r`" at the end of the first sentence, and remove the last sentence.